### ISSUE TITLE
VB: VisualBasicAddParameterCodeFixProvider should not crash when code has semantic errors

### ISSUE THREAD

**Issue by ericmutta** [dotnet/roslyn#23927]
**Version Used**: 

Visual Studio Community 2017 v15.5.1

**Steps to Reproduce**:

Type in the following code in a VB console project:

```vb
Public Module Module1
  Private Class Foo
    Public Sub New(Arg1 As Integer)
    End Sub

    Public Sub New(Arg1 As Integer, Arg2 As Integer)
    End Sub
  End Class

  Public Sub Main()
    Dim x = New Foo(, 0) '<--- this line has an error (first argument not specified)
  End Sub
End Module
```

The editor will correctly put a red squiggle on `Foo` and complain than an argument has not been specified in that constructor call, but when you hover over the squiggle to see the error, the `VisualBasicAddParameterCodeFixProvider` is triggered and crashes:

![vs_addparamprovider_crash](https://user-images.githubusercontent.com/20465797/34341964-d48b5a3c-e9b3-11e7-8fb2-258970770c0d.png)

If you click the Enable button in that error message near the top, then hover over the red squiggly again it crashes again. You can repeat this as many times as you want and it crashes every time. The stack trace given is:

```
System.ArgumentNullException : Value cannot be null.
Parameter name: node
   at Microsoft.CodeAnalysis.VisualBasic.VBSemanticModel.GetTypeInfoForNode(SyntaxNode node,CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.VisualBasic.VBSemanticModel.GetTypeInfoCore(SyntaxNode node,CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.AddParameter.AbstractAddParameterCodeFixProvider`6.DetermineFirstArgumentToAdd(SemanticModel semanticModel,ISyntaxFactsService syntaxFacts,StringComparer comparer,IMethodSymbol method,SeparatedSyntaxList`1 arguments,TArgumentSyntax argumentOpt)
   at async Microsoft.CodeAnalysis.AddParameter.AbstractAddParameterCodeFixProvider`6.HandleObjectCreationExpressionAsync[TArgumentSyntax,TAttributeArgumentSyntax,TArgumentListSyntax,TAttributeArgumentListSyntax,TInvocationExpressionSyntax,TObjectCreationExpressionSyntax](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.AddParameter.AbstractAddParameterCodeFixProvider`6.RegisterCodeFixesAsync[TArgumentSyntax,TAttributeArgumentSyntax,TArgumentListSyntax,TAttributeArgumentListSyntax,TInvocationExpressionSyntax,TObjectCreationExpressionSyntax](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.PerformActionAsync(<Unknown Parameters>)
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
```

**Expected Behavior**:

It should not crash. This seems similar to #23807 and it appears as if these code fix providers don't have the same resilience of the core compilers. In both issues, the compiler reports the syntax/semantic errors without crashing, it would be great if the code fix providers could leverage that resilience in their analysis before triggering 👍

CC: @jcouv @dpoeschl 
PS: Merry Christmas!

**Comment by jcouv** [comment]
I think this code fixer should not trigger on that error. Started PR for fix.
Thanks and Merry Christmas to you too!

**Comment by ericmutta** [comment]
@jcouv the speed with which you fix these issues (even during global holidays!) never fails to amaze! Thanks a lot for your efforts 👍

**Comment by jcouv** [comment]
Correction: the fixer should handle this error (30445). Our test suite caught a regression with my initial fix. PR updated.

``` vb
Class Test
    Sub Main()
        Dim a = New A() ' BC30445 should trigger the fix here
    End Sub
End Class
Class A
    Sub New(x As Integer)
    End Sub
End Class
```

**Comment by jcouv** [comment]
Fix merged for 15.6 preview 3.

**Comment by ericmutta** [comment]
> @jcouv Fix merged for 15.6 preview 3.

Absolutely awesome, thanks again for fixing this so fast and I look forward to 15.6!
